src/admin.tsx

```diff
@@ -2,60 +2,153 @@ import React, { useContext, useEffect, useState } from 'react';
+    });
+  }
 
 
   const displayTreeComponent = () => {
     if (!data) {
       return;
     }
 
-    // Group by subcategory and all it's videos
-    const newData: Record<string, Record<string, Ambiance[]>> = {};
-    data?.forEach((category) => {
-      const groupedVideos: Record<string, Ambiance[]> = {};
-      category.videos.forEach((vid) => {
+    // Map of documentId -> (Map of subcategory name -> videos in subcategory)
+    const newData: AmbianceDisplayType = {};
+    Object.entries(data).forEach(([documentId, value]) => {
+
+      // Map of subcategory name -> videos in subcategory
+      const groupedVideos: SubcategoryMap = {};
+      value.videos.forEach((vid) => {
+        // If subcategory already exists, concat to existing list of videos
         if (groupedVideos[vid.group]) {
           groupedVideos[vid.group] = groupedVideos[vid.group].concat(vid);
-        } else {
+        } else { // Create new key/value in map, with array of one video
           groupedVideos[vid.group] = [vid];
         }
       });
 
-      newData[category.name] = groupedVideos;
+      // Add grouped videos to newData map using documentId as key
+      newData[documentId] = {
+        friendlyName: value.name,
+        videosBySubcategory: groupedVideos
+      };
     })
 
-    const elements = Object.entries(newData).map(([key, groupedVideos]) => (
-      <TreeItem className='tree-item' key={key} nodeId={key} label={key} sx={{ textAlign: 'left' }}>
-        {Object.entries(groupedVideos).map(([subcategory, vids]) => (
-          <TreeItem key={key + subcategory} nodeId={key + subcategory} label={subcategory} sx={{ textAlign: 'left' }}>
-            {vids.map((vid) => (
-              <TreeItem key={vid.name} nodeId={vid.name} label={vid.name} sx={{ textAlign: 'left' }} />
+    const addIcon = (
+      <Tooltip title="Add Video">
+        <IconButton color="secondary" size="small">
+          <AddIcon />
+        </IconButton>
+      </Tooltip>
+    );
+
+    // key = ambiance category, i.e., animalCrossing, bg3
+    const elements = Object.entries(newData).map(([documentId, ambianceDisplay], index) => (
+      <TreeItem className='tree-item' key={documentId} nodeId={documentId} label={ambianceDisplay.friendlyName} sx={{ textAlign: 'left' }}>
+        {Object.entries(ambianceDisplay.videosBySubcategory).map(([subcategory, vids]) => (
+          <TreeItem key={documentId + subcategory} nodeId={documentId + subcategory} label={subcategory} sx={{ textAlign: 'left' }}>
+            {vids.map(vid => (
+              <TreeItem collapseIcon={<BuildCircleIcon className='build-icon' />} expandIcon={<BuildCircleIcon />} key={vid.name} nodeId={vid.name} label={vid.name} sx={{ textAlign: 'left' }} >
+                <VideoEditor videoUrlCode={vid.code} video={vid} documentId={documentId} editVideo={editVideo} deleteVideo={deleteVideo} />
+              </TreeItem>
             ))}
+            {/* add new collapse icon, add tooltip to edit icon above */}
```

Comment:
Can we do this in this PR? Dont think we need the new collapse icon anymore, but maybe a tooltip to say "Edit video" would be nice similar to the "Add video" tooltip.

Reply:
Oh nice, even more incentive to create a small re-usable component from my comment above. With props for the tooltip text and icon?